sarathi/model_executor/model_runner.py

```diff
@@ -218,21 +221,23 @@ def profile_num_available_blocks(
     def run(
         self,
         seq_metadata_list: List[SequenceMetadata],
-        gpu_cache: Optional[List[torch.Tensor]] = None,
     ) -> torch.Tensor:
         # Prepare input tensors.
         with self._prepare_inputs_e2e_timer:
             input_tokens, input_positions = self._prepare_inputs(seq_metadata_list)
 
-        get_attention_wrapper().begin_forward(seq_metadata_list)
+        self.attention_backend_wrapper.begin_forward(seq_metadata_list)
 
         with self._model_execution_e2e_timer:
             # Execute the model.
             try:
+                
+                #TODO (vkomperla3): Make attention backend changes for all Model types (falcon, internlm, etc.)
```

Comment:
To Do before merge:

Make attention backend changes to all model classes (Falcon, InternLM, etc.). Currently, changes have only been made for Llama